neutron/neutron.go

```diff
@@ -51,12 +58,45 @@ func New(client client.Client) *Client {
 	return &Client{client}
 }
 
-// ListNetworksV2 gives details on available networks
-func (c *Client) ListNetworksV2() ([]NetworkV2, error) {
+// ----------------------------------------------------------------------------
+// TODO(hml): copied from the nova package.  However it should really be pulled out
+// and shared between goose pkgs, however we don't want to break compatibility at this time.
+//
+// Filtering helper.
+//
+// Filter builds filtering parameters to be used in an OpenStack query which supports
+// filtering.  For example:
+//
+//     filter := NewFilter()
+//     filter.Set(neutron.FilterRouterExternal, "true")
+//     resp, err := neutron.ListNetworks(filter)
+//
+type Filter struct {
+	v url.Values
+}
+
+// NewFilter creates a new Filter.
+func NewFilter() *Filter {
+	return &Filter{make(url.Values)}
+}
+
+func (f *Filter) Set(filter, value string) {
+	f.v.Set(filter, value)
+}
+
+// ----------------------------------------------------------------------------
+
+// ListNetworksV2 gives details on available networks, zero or one Filters
```

Comment:
why not just have it take a `*Filter`, which may be nil? that would make it the same as other methods in goose

is this to avoid bumping the version? TBH I think it might be time anyway. is it to onerous to bump to say, v2-unstable, and then we can iron out the kinks over a period of time and then call it v2 (but still use v2-unstable in juju along the way)?